lms/djangoapps/courseware/user_state_client.py

```diff
@@ -24,6 +24,21 @@
 class DjangoXBlockUserStateClient(XBlockUserStateClient):
     """
     An interface that uses the Django ORM StudentModule as a backend.
+
+    A note on the format of state storage:
+        The state for an xblock is stored as a serialized JSON dictionary. The model
+        field that it is stored in can also take on a value of ``None``. To preserve
+        existing analytic uses, we will preserve the following semantics:
+
+        A state of ``None`` means that the user hasn't ever looked at the xblock.
+        A state of ``"{}"`` means that the XBlock has at some point stored state for
+           the current user, but that that state has been deleted.
+        Otherwise, the dictionary contains all data stored for the user.
```

Comment:
Good note - but ugh...
(ugh = Whether or not the stored state is non-None shouldn't depend on whether a user has simply looked at a problem - but whether they've answered it.)